File: 4252-FirstUniqueEvenElement/4252-FirstUniqueEvenElement.java
// Last updated: 3/30/2026, 2:10:21 PM
class Solution {
    public int firstUniqueEven(int[] nums) {
        HashMap<Integer,Integer> map = new LinkedHashMap<>();
        for(int num : nums) map.put(num,map.getOrDefault(num,0)+1);
        for(int key : map.keySet()){
            if(key%2==0 && map.get(key)==1) return key;
        }
        return -1;
    }
}

File: 4252-FirstUniqueEvenElement/4252-FirstUniqueEvenElement.cpp
// Last updated: 3/30/2026, 2:10:27 PM
class Solution {
public:
    int firstUniqueEven(vector<int>& nums) {
        unordered_map<int,int> hashmap;
        for(int num : nums) hashmap[num]+=1;
        for(int num : nums){
            if(hashmap.find(num) != hashmap.end()){
                if(num%2==0 && hashmap[num]==1) return num;
            }
        }
        return -1;
    }
};